lib/parser/source/buffer.rb

```diff
@@ -95,9 +95,7 @@ def self.reencode_string(input)
         elsif detected_encoding == Encoding::BINARY
           input
         else
-          input.
-            force_encoding(detected_encoding).
-            encode(Encoding::UTF_8)
+          input.force_encoding(detected_encoding)
```

Comment:
This will fail if a file with a non-ASCII-compatible encoding is ever parsed. Or something with ShiftJIS, IIRC.